### PR TITLE
fix: remove invalid children for steps

### LINKED ISSUE #2329: steps may have null children

### PR DESCRIPTION
**Pull request by likezero**
fix #2329

* [x] Make sure that you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
* [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
* [x] Rebase before creating a PR to keep commit history clear.
* [x] Add some descriptions and refer relative issues for you PR.

---
This change is [Reviewable](https://reviewable.io/reviews/ant-design/ant-design-mobile/2562)

### CODE REVIEW

File: components/steps/index.tsx [hunk 1]
```diff
@@ -42,13 +41,21 @@ export default class Steps extends React.Component<StepsProps, any> {
     this.stepRefs = [];
     const { children, status, size } = this.props;
     const current = this.props.current as number;
+
     // flattern the array at first https://github.com/ant-design/ant-design-mobile/issues/934
-    let newChildren: any[] = React.Children.map(children, item => item);
-    newChildren = React.Children.map(newChildren, (item: any, index) => {
+    const filterChildren: any[] = [];
+    if (children && children.length) {
+      children.forEach( (item: any) => {
```
Comment (warmhug) [inline components/steps/index.tsx]: 多了个 空格，lint 过了吗？

Comment (likezero) [reply components/steps/index.tsx]: 本地过了，push上来之后也会跑lint吧

**Comment by likezero** [comment]
```jsx
<Steps size="small" current={1}>
      {11}
      {null}
      {true}
      {false}
      {'string'}
      <Step title="Finished" />
      <Step title="In Progress" />
      <Step title="Waiting" />
 </Steps>
```
前5个类型的元素均会被过滤掉。

File: components/steps/index.tsx [hunk 2]
```diff
@@ -42,13 +41,21 @@ export default class Steps extends React.Component<StepsProps, any> {
     this.stepRefs = [];
     const { children, status, size } = this.props;
     const current = this.props.current as number;
+
     // flattern the array at first https://github.com/ant-design/ant-design-mobile/issues/934
-    let newChildren: any[] = React.Children.map(children, item => item);
-    newChildren = React.Children.map(newChildren, (item: any, index) => {
+    const filterChildren: any[] = [];
```
Comment (warmhug) [inline components/steps/index.tsx]:
```
<Steps>
  <Step title="第一步" />
  <Step title="第二步" />
  {
    Array(3).fill(1).map(v => (
      <Step key={Math.random()} title="第三步" />
   ))
  }
</Steps>
```
这种用法情况测试 ok 吗？#934

Comment (likezero) [reply components/steps/index.tsx]: 我试试

Comment (likezero) [reply components/steps/index.tsx]: 测试了一下，是OK的
image

File: components/steps/index.tsx (hunk 1 above)
Comment (warmhug) [reply components/steps/index.tsx]: 按说不应该过的，多余空格去了吧

Comment (likezero) [reply components/steps/index.tsx]: 去掉了

**Comment by warmhug** [comment]
可以了就直接合了吧

**Pull request merged**